dora/src/typeck/expr.rs

```diff
@@ -1262,7 +1264,11 @@ impl<'a, 'ast> TypeCheck<'a, 'ast> {
                 CallType::Trait(trait_id, fct_id)
             } else {
                 let method_type = lookup.found_class_type().unwrap();
-                CallType::Method(method_type, fct_id, type_params.clone())
+                if method_type.is_module() {
+                    CallType::ModuleMethod(method_type, fct_id, type_params.clone())
```

Comment:
We should have tests for module methods with type arguments as well. It would also be great to test a module for a generic class as well.

Reply:
I have an uncommitted `Assert` module that would work perfectly for this ...

Reply:
I think I've already mentioned that we should try not to use stdlib functions in tests. Better to have a separate test specifically written for this.

Reply:
Understood, I added a minimized test to typeck!